hs/src/Reach/AST/Base.hs

```diff
@@ -171,24 +171,26 @@ data CompilationError = CompilationError
   , ce_position :: [Int]
   , ce_offendingToken :: Maybe String
   , ce_errorCode :: String
+  , ce_stackTrace :: String
   }
   deriving (Show, Generic, ToJSON, FromJSON)
 
-makeCompilationError :: (ErrorSuggestions a, ErrorMessageForJson a, Show a, HasErrorCode a) => SrcLoc -> a -> CompilationError
-makeCompilationError src err =
+makeCompilationError :: (ErrorSuggestions a, ErrorMessageForJson a, Show a, HasErrorCode a) => SrcLoc -> a -> [String] -> CompilationError
+makeCompilationError src err stackTrace =
   CompilationError
     { ce_suggestions = snd $ errorSuggestions err
     , ce_offendingToken = fst $ errorSuggestions err
     , ce_errorMessage = errorMessageForJson err
     , ce_position = srcloc_line_col src
     , ce_errorCode = makeErrCode (errPrefix err) (errIndex err)
+    , ce_stackTrace = unwords stackTrace
```

Comment:
This might be more useful in the future if the field were `string[]` instead of `string`, which should be valid JSON. If keeping `string`, I think `unlines` makes more sense than `unwords`

Reply:
I do want to keep `ce_stackTrace` a `string`, because then I can just do `String.match` on `ce_stackTrace`, instead of
```ts
ce_stackTrace.forEach(line => {
  const someRegEx = /.+/;
  const result = line.match(someRegEx);
  if (result) {
    // do something
  }
});
```

I think the approach using `unwords` is cleaner, at least for the time being. The Regular Expression on the TypeScript side appears to work more nicely when `ce_stackTrace` isn't joined using `\n`.

I'll keep `unlines` in mind for future reference, though. Thank you for your input!